cluster-lifecycle/update-managedcluster-status-to-include-clusterinfo.md

```diff
@@ -0,0 +1,254 @@
+
+It is required from other components that more cluster specific info should be added into ManagedCluster API. This information includes the clusterID, the ocp distribution version, vendor and cloud provider etc. On another hand, ManagedCluster API should be designed to have a generic and extensible structure to include this info that also can work with different implementations.
+
+### Goals
+
+- Allow extensible info in managed cluster to be added in ManagedCluster API
+- Remove cluster info lables from managed cluster eventually and use managed cluster status instead;
+- Align wth upstream design;
+
+### Non-Goals
+
+- Replace and deprecate ManagedClusterInfo API
+
+## Proposal
+
+To align with the upstream proposal for ClusterID (https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit), leverage ClusterClaims created on managed cluster to store and manage cluster information.
+
+### Risks and Mitigations
+
+- The upstream proposal for ClusterID is not finalized yet and may change in the future.  
+The mitigation is to create ClusterClaim CRD in group cluster.open-cluster-management.io with version v1alph1 for now. Once the proposal is finalized, support both of them for one or two releases and then deprecate the one in group cluster.open-cluster-management.io.
+
+## Design Details
+
+As shown in the table below, ManagedClusterInfo includes 10 fields in its status. Some of them will NOT be merged into ManagedCluster.
+
+| Name    | Description    | Comments   |
+| ---- | ---- | ---- |
+| Conditions | condition information for a managed cluster | **Not merged**. ManagedCluster has its own Conditions. |
+| Version | the kube version of managed cluster | **Not merged**. ManagedCluster has a field with same name in its status, which also points to the kube version of managed cluster |
+| KubeVendor | the kubernetes provider of the managed cluster| |
+| CloudVendor | the cloud provider for the managed cluster | |
+| ClusterID | the identifier of managed cluster | |
+| DistributionInfo | the information about distribution of managed cluster | **Not merged**. It's product specific. |
+| ConsoleURL | the url of console in managed cluster | ？ |
+| NodeList | a list of the status of nodes | **Not merged**. There is a scale issue for a cluster with a large number of nodes. |
+| LoggingEndpoint | the endpoint to connect to logging server of managed cluster | **Not merged**. It's product specific. |
+| LoggingPort | the port to connect to logging server of managed cluster | **Not merged**. It's product specific. |
+
+ClusterClaim is a CRD which comes from the upstream proposal for ClusterID. It’s a cluster-scoped resource created on managed clusters and provides a way to store cluster scoped information. Its name is a well known or custom name to identify the claim. The following shows the definition of ClusterClaim.
```

Comment:
So, does `ClusterID` in this doc refer to the upstream concept? OpenShift clusters already have a unique ID coordinate, do they not?

Reply:
Yes, ocp already have one. The original thought is that we have another agent to create id claim with ocp clusterID  in it. Thus we have an ocp's implementation for clusterID. But after reading the upstream concept again. I feel that clusterID will be part of the dns domain generated by MCS api. So the upstream clusterID concept (id.k8s.io) might align more with name of managedcluster which is more user readable and dns-label conformant. Should we consider having name of managedcluster as the value of id.k8s.io clusterclaim, and having another claim `id.openshift.io` for the clusterID in openshift?

Reply:
I think having two different claims makes sense.